### ISSUE TITLE
Ignore TS checks

### ISSUE THREAD

**Issue by jdussouillez** [vojtechhabarta/typescript-generator#429]
Since the introduction of generic type tagged unions, I have an error when compiling the TS because of the [`noUnusedParameters`](https://www.typescriptlang.org/docs/handbook/compiler-options.html) option.

Code generated with [v2.17.564](https://github.com/vojtechhabarta/typescript-generator/releases/tag/v2.17.564) :

```typescript
// ...
export namespace com.test.tsgen {

    export type EntityUnion<T> = com.test.tsgen.Foo | com.test.tsgen.Bar;

}
```

If you have the `noUnusedParameters` option activated, the TS compilation fails with the following error: `model.ts(44,29): error TS6133: 'T' is declared but its value is never read.`.

**It's not a typescript-generator issue**, but to prevent this, would it be possible to add a `@ts-ignore` comment on the generic type unions ?

```typescript
// ...
export namespace com.test.tsgen {

    // @ts-ignore
    export type EntityUnion<T> = com.test.tsgen.Foo | com.test.tsgen.Bar;

}
```

## Code to reproduce

*The full code to generate this `model.ts` is in https://github.com/vojtechhabarta/typescript-generator/issues/428*

### model.ts (generated)
```typescript
/* tslint:disable */
/* eslint-disable */
// Generated using typescript-generator version 2.17-SNAPSHOT on 2019-10-24 08:46:07.

export namespace com.test.tsgen {
    export class Entity<T> {
        "@type": "Bar" | "Foo";
        id?: T;
        entityType?: com.test.tsgen.EntityType;
    }
}

export namespace com.test.tsgen {
    export class Bar extends com.test.tsgen.Entity<number> {
        "@type": "Bar";
        id?: number;
    }
}

export namespace com.test.tsgen {
    export class EntityCollection {
        entities?: com.test.tsgen.EntityUnion<any>[];
    }
}

export namespace com.test.tsgen {
    export class Foo extends com.test.tsgen.Entity<string> {
        "@type": "Foo";
        id?: string;
    }
}

export namespace com.test.tsgen {
    export type EntityUnion<T> = com.test.tsgen.Foo | com.test.tsgen.Bar;
}

export namespace com.test.tsgen {
    export enum EntityType {
        FOO = "FOO",
        BAR = "BAR",
    }
}
```

#### tsconfig.config
```js
{
    "compileOnSave": false,
    "compilerOptions": {
        "baseUrl": "./",
        "outDir": "./dist/out-tsc",
        "sourceMap": true,
        "declaration": false,
        "module": "esnext",
        "moduleResolution": "node",
        "emitDecoratorMetadata": true,
        "experimentalDecorators": true,
        "noUnusedLocals": true,
        "noUnusedParameters": true, // <-- causes the issue
        "noImplicitReturns": true,
        "noImplicitAny": true,
        "alwaysStrict": true,
        "target": "es2015",
        "lib": [
            "es2015"
        ]
    },
    "files": [
        "model.ts"
    ]
}
```

#### Commande line
```console
$ ls
model.ts tsconfig.json
$ tsc
model.ts(44,29): error TS6133: 'T' is declared but its value is never read.
```

**Comment by jdussouillez** [comment]
A better solution would be to add `// @ts-nocheck` ([Typescript 3.7](https://devblogs.microsoft.com/typescript/announcing-typescript-3-7-beta/)) at the begging of the file (under `/* eslint-disable */`) to ignore all TS checks for the generated file.

This comment is ignored with TS < 3.7 so it would not cause any issues with previous versions.

**Comment by vojtechhabarta** [comment]
Are you compiling generated file together with your source files? Would it be possible to compile it separately with different compiler options?

Or would it be acceptable for you to check unused parameters using ESLint instead of TypeScript compiler with `noUnusedParameters` option? I don't consider this kind of issues being errors so I think ESLint is better tool for checking it. And ESLint is already disabled in generated files.

**Comment by jdussouillez** [comment]
Yes I'm compiling the generated file with the other source files. It would be possible to have another `tsconfig.json` to compile just the generated file but I prefer not to.

I'm using TSLint now and the [`no-unused-variable` is deprecated because Typescript does it too](https://github.com/palantir/tslint/blob/master/CHANGELOG.md#warning-deprecations).
I will change my linter to ESLint soon because [TSLint is deprecated](https://medium.com/palantir/tslint-in-2019-1a144c2317a9) so my issue will be fixed, but I can't upgrade the typescript-generator until I change to ESLint.

**Comment by vojtechhabarta** [comment]
@jdussouillez I am going through older issues, would you be still interested is some switch for `// @ts-nocheck` comment?

**Comment by jdussouillez** [comment]
@vojtechhabarta Yes I am.

This issue prevents me to upgrade from [v2.15.527](https://github.com/vojtechhabarta/typescript-generator/releases/tag/v2.15.527) to the latest version because I don't want to disable the `noUnusedParameters` of my TS options. At least not until I migrate from TSLint to ESLint ([I'm waiting for ESLint to implement some missing features with the TS plugin](https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/ROADMAP.md))

Even if we put aside the TS/ESlint thing, adding an option to tell the TS compiler to ignore a generated file is IMO a good thing to do.

**Comment by vojtechhabarta** [comment]
Released in [v2.21.588](https://github.com/vojtechhabarta/typescript-generator/releases/tag/v2.21.588).
